p2p/discovery/addrbook.go

```diff
@@ -106,16 +107,16 @@ type addrBook struct {
 
 // AddOurAddress adds one of our addresses.
 func (a *addrBook) AddLocalAddress(addr *node.Info) {
-	a.mtx.Lock()
+	a.localAddrMtx.Lock()
 	a.localAddresses = append(a.localAddresses, addr)
-	a.mtx.Unlock()
+	a.localAddrMtx.Unlock()
 }
 
 // IsLocalAddress returns true if this address was added as a local address before.
 func (a *addrBook) IsLocalAddress(addr *node.Info) bool {
```

Comment:
since you changed mutexs, every occurrence of `isLocalAddressUnlocked` should be now using `IsLocalAddress`

Reply:
yep, you are right, my fault